# Costmap parameters
global_frame: /map
robot_base_frame: /base_link
transform_tolerance: 0.2
rolling_window: false
track_unknown_space: true
always_send_full_costmap: true
width: 7.0
height: 4.0
resolution: 0.05
origin_x: -3.5
origin_y: -2.0
publish_cost_grid: true
update_frequency: 5.0
publish_frequency: 2.0

footprint_padding: 0.01

plugins:
- {name: priority_layer,       type: "user_priority_map/UserPriorityLayer"}
- {name: obstacle_layer,       type: "layered_costmap_2d/ObstacleLayer"}
#- {name: activity_layer,       type: "dynamic_map/ActivityLayer"}

activity_layer:
  transform_tolerance: 0.2  
  global_frame: /map
  robot_base_frame: /base_link
  map_type: costmap 
  track_unknown_space: true
  publish_cost_grid: true
  resolution: 0.05
  publish_voxel_map: false
  # sensors
  max_obstacle_height: 2.0
  obstacle_range: 50.0
  raytrace_range: 50.0  

  observation_sources: base_scan0 base_scan1
  base_scan0: {
      data_type: LaserScan,
      sensor_frame: /base_laser_link0,
      topic: /base_scan0,
      expected_update_rate: 0.0,
      observation_persistence: 0.0,
      marking: true,
      clearing: true,
      min_obstacle_height: 0.0
  }

  base_scan1: {
      data_type: LaserScan,
      sensor_frame: /base_laser_link1 ,
      topic: /base_scan1,
      expected_update_rate: 0.0,
      observation_persistence: 0.0,
      marking: true,
      clearing: true,      
      min_obstacle_height: 0.0
  } 

priority_layer:
  map_topic: /map
  first_map_only: true
  subscribe_to_updates: false

obstacle_layer:
  transform_tolerance: 0.2  
  global_frame: /map
  robot_base_frame: base_link
  map_type: costmap 
  track_unknown_space: true
  resolution: 0.05
  publish_voxel_map: false
  # sensors
  max_obstacle_height: 2.0
  obstacle_range: 50.0
  raytrace_range: 50.0  


  observation_sources: base_scan0 base_scan1
  base_scan0: {
      data_type: LaserScan,
      sensor_frame: /base_laser_link0,
      topic: base_scan0,
      expected_update_rate: 0.0,
      observation_persistence: 0.0,
      marking: true,
      clearing: true,
      min_obstacle_height: 0.0
  }

  base_scan1: {
      data_type: LaserScan,
      sensor_frame: /base_laser_link1,
      topic: base_scan1,
      expected_update_rate: 0.0,
      observation_persistence: 0.0,
      marking: true,
      clearing: true,      
      min_obstacle_height: 0.0
  } 

